src/VolumeRenderSettings.ts

```diff
@@ -137,4 +137,105 @@ export class VolumeRenderSettingUtils {
     renderSettings.emissive = new Array(volume.num_channels).fill([0, 0, 0]);
     renderSettings.glossiness = new Array(volume.num_channels).fill(0);
   }
+
+  /**
+   * Recursively compares two arrays.
+   * Non-array elements are compared using strict equality comparison.
+   */
+  private static compareArray(a1: unknown[], a2: unknown[]): boolean {
+    if (a1.length !== a2.length) {
+      return false;
+    }
+    for (let i = 0; i < a1.length; i++) {
+      const elem1 = a1[i];
+      const elem2 = a2[i];
+      if (elem1 instanceof Array && elem2 instanceof Array) {
+        if (!this.compareArray(elem1, elem2)) {
+          return false;
+        }
+      } else if (elem1 !== elem2) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Compares two VolumeRenderSettings objects.
+   * @returns true if both objects have identical settings.
+   */
+  public static isEqual(o1: VolumeRenderSettings, o2: VolumeRenderSettings): boolean {
+    for (const key of Object.keys(o1)) {
+      const v1 = o1[key];
+      const v2 = o2[key];
+      if (v1 instanceof Array) {
+        if (!this.compareArray(o1[key], o2[key])) {
+          return false;
+        }
+      } else if (v1 && v1.bmin !== undefined) {
+        // Bounds object
+        const bounds1 = v1 as Bounds;
+        const bounds2 = v2 as Bounds;
+        if (!bounds1.bmin.equals(bounds2.bmin) || !bounds1.bmax.equals(bounds2.bmax)) {
+          return false;
+        }
+      } else if (v1 instanceof Vector3 || v1 instanceof Vector2 || v1 instanceof Euler) {
+        if (!v1.equals(o2[key])) {
+          return false;
+        }
+      } else {
+        // number, boolean, string
+        if (v1 !== o2[key]) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Recursively creates and returns a deep copy of an array.
+   * Note: assumes values in the array are either primitives (numbers) or arrays of primitives.
+   */
+  private static deepCopyArray(a: unknown[]): unknown[] {
+    const b: unknown[] = new Array(a.length);
+    for (let i = 0; i < a.length; i++) {
+      const val = a[i];
+      if (val instanceof Array) {
+        b[i] = this.deepCopyArray(val);
+      } else {
+        b[i] = val;
+      }
+    }
+    return b;
+  }
+
+  /**
+   * Creates a deep copy of a VolumeRenderSettings object.
+   * @param src The object to create a clone of.
+   * @returns a new VolumeRenderSettings object with identical fields that do not
+   * share references with the original settings object.
+   */
+  public static clone(src: VolumeRenderSettings): VolumeRenderSettings {
+    const dst = {};
+    for (const key of Object.keys(src)) {
+      const val = src[key];
+      if (val instanceof Array) {
+        dst[key] = this.deepCopyArray(val);
+      } else if (key === "bounds") {
+        // Bounds
+        dst["bounds"] = { bmax: new Vector3(), bmin: new Vector3() } as Bounds;
```

Comment:
i'd use `dst.bounds` here and make sure it's typed to VolumeRenderSettings

Reply:
also, can you use instanceof Bounds here instead of a string comparison ?

Reply:
Ah, I can't do an `instanceof Bounds` comparison because it's a type alias and not an actual class :((
> `'Bounds' only refers to a type, but is being used as a value here.ts(2693)`

Reply:
I could check to see if the value is an object and if .bmax and .bmin exist on it? But that seems like it could run into the same problem if the names of `bmin/bmax/bounds` every changes

Reply:
yeah, if Bounds were a class it would work.  it's either change Bounds to be a class or leave it as is for now.  not a show stopper.

Reply:
Going to leave it for now-- I tried changing Bounds to a class but there seem to be some consequences that cause other things to break (I suspect there are instances of Bounds not being declared correctly). Fortunately one of the tests for the VolumeRenderSettings will catch if `bmin/bmax/bounds` breaks or if the properties change, so I think this is okay.